src/DotNet/Emit/DynamicMethodBodyReader.cs

```diff
@@ -534,6 +537,16 @@ class ExceptionInfo {
 			return tokens[(int)index];
 		}
 
+		/// <inheritdoc />
+		public override void RestoreMethod(MethodDef method) {
+			base.RestoreMethod(method);
+
+			var body = method.Body;
+			if (initLocals.HasValue)
+				body.InitLocals = initLocals.Value;
+			body.MaxStack = (ushort)Math.Min(maxStack, ushort.MaxValue);
```

Comment:
I added the code to initialize MaxStack so that when the using the preserve maxstack metadata writer option is used the max stack is still a proper value. The reason why for the cast is that I copied the line from https://github.com/0xd4d/dnlib/pull/470/files#diff-5a6271fc9c1b45930fcd380798359844d3431c2c3f0b48a56fa8325f44975edcR395